pingcap/tidb-lightning
# Introduce a basic web interface

### What problem does this PR solve? 

Introduce a web interface for viewing the current task progress and perform some task management in server mode (submit task, cancel task, pause and resume, etc).

### What is changed and how it works?

* The web interface is a Single Page Application written using React with Material-UI. 
* It communicates with Lightning via simple HTTP API. Updating is done by regular refreshing.
* The entire checkpoint of the current task is cloned for the web interface to view.
* (Added "pause"/"resume" and various task management APIs to make the web interface easier to demonstrate)

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
    * Manually execute the tasks and observe the change in the web interface

Side effects

 - Possible performance regression
 - Increased code complexity

Related changes

 - Need to update the documentation
 - Need to update the `tidb-ansible` repository
 - Need to be included in the release note
